src/action/base/create_or_merge_nix_config.rs

```diff
@@ -0,0 +1,490 @@
+                .map_err(|e| ActionError::Custom(Box::new(e)))?;
+            this.nix_configs.existing_nix_config = Some(existing_nix_config.clone());
+            let mut merged_nix_config = NixConfig::new();
+            let mut unmergeable_config_names = Vec::new();
+
+            for (pending_conf_name, pending_conf_value) in pending_nix_config.settings() {
+                if let Some(existing_conf_value) =
+                    existing_nix_config.settings().get(pending_conf_name)
+                {
+                    let pending_conf_value = pending_conf_value.split(' ').collect::<Vec<_>>();
+                    let existing_conf_value = existing_conf_value.split(' ').collect::<Vec<_>>();
+
+                    if pending_conf_value
+                        .iter()
+                        .all(|e| existing_conf_value.contains(e))
+                    {
+                        // If _all_ the values we want are present in the existing config,
+                        // merged_nix_config will be empty and this will be marked as completed. We
+                        // don't return early here because there may be more config options to
+                        // check.
+                    } else if MERGEABLE_CONF_NAMES.contains(&pending_conf_name.as_str()) {
+                        let mut merged_conf_value = Vec::with_capacity(
+                            pending_conf_value.len() + existing_conf_value.len(),
+                        );
+                        merged_conf_value.extend(pending_conf_value);
+                        merged_conf_value.extend(existing_conf_value);
+                        merged_conf_value.dedup();
+                        let merged_conf_value = merged_conf_value.join(" ");
+
+                        merged_nix_config
+                            .settings_mut()
+                            .insert(pending_conf_name.to_owned(), merged_conf_value.to_owned());
+                    } else {
+                        unmergeable_config_names.push(pending_conf_name.to_owned());
+                    }
+                } else {
+                    merged_nix_config
+                        .settings_mut()
+                        .insert(pending_conf_name.to_owned(), pending_conf_value.to_owned());
+                }
+            }
+
+            if !unmergeable_config_names.is_empty() {
+                return Err(ActionError::Custom(Box::new(
+                    CreateOrMergeNixConfigError::UnmergeableConfig(
+                        unmergeable_config_names,
+                        this.path,
+                    ),
+                )));
+            }
+
+            if !merged_nix_config.settings().is_empty() {
+                this.nix_configs.merged_nix_config = merged_nix_config;
+                return Ok(StatefulAction::uncompleted(this));
+            }
+
+            tracing::debug!(
+                "File `{}` already contains what we want",
```

Comment:
```suggestion
                "Setting Nix configurations in `{}` already complete",
```

We should match https://github.com/DeterminateSystems/nix-installer/blob/9c8284cd0626ccdcdba709f6ade4faa604fd0f79/src/action/base/create_file.rs#L126 https://github.com/DeterminateSystems/nix-installer/blob/9c8284cd0626ccdcdba709f6ade4faa604fd0f79/src/action/base/create_or_insert_into_file.rs#L129 and it's ilk.